packages/store/addon/-private/system/core-store.ts

```diff
@@ -1142,7 +1142,8 @@ abstract class CoreStore extends Service {
         if (IDENTIFIERS) {
           // ensure that regardless of id returned we assign to the correct record
           if (payload.data && !Array.isArray(payload.data)) {
-            payload.data.lid = identifier.lid;
+            payload = { ...payload };
+            payload.data = { ...payload.data, lid: identifier.lid };
```

Comment:
not sure how I feel about this. While it does assign to the passed in payload (a mutation) this is a hot path where making an immutable copy is likely worse.

Reply:
Discussed on Discord. We will open an issue and remove the bugfix from this PR.